torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -456,9 +459,133 @@ void ONNXAssignOutputShape(
   }
 }
 
+// Check if subblock output is output of prim::Uninitialized,
+// or output of prim::Uninitialized->onnx::Identity
+bool IsUninitializedBlockOutput(Node* n) {
+  if (n->kind() == ::c10::onnx::Identity &&
+      n->inputs()[0]->node()->kind() == prim::Uninitialized)
+    return true;
+  if (n->kind() == prim::Uninitialized)
+    return true;
+  return false;
+}
+
+Node* CreateCastToBoolNode(Value* val, Graph* graph) {
```

Comment:
should we handle the issue for Input to If node not being Bool in erase_number_types pass or is the cast to Bool the only solution?

Reply:
Why erase_number_types pass? Is there an issue with this pass that need to be addressed?

Reply:
erase_numeber_types pass cast bool constants to Long, and those constants can be inputs to if node.